Filer.py

```diff
@@ -312,6 +329,70 @@ def make_tree(rel, pathname, clean_expired=True):
     return tree
 
 
+def read_users(htpasswd_file):
```

Comment:
This is exactly the kind of code complexity I tried to avoid. Working on single files in the User's directory makes for a very simple to understand code: Just ask the file system if the file is there, done. With that added complexity, we would introduce more room for inconsistencies and hard to read and understand constructs like the one below: